libs/template/templates/default-python/template/{{.project_name}}/src/{{.project_name}}/main.py.tmpl

```diff
@@ -1,16 +1,12 @@
-{{- /*
-We use pyspark.sql rather than DatabricksSession.builder.getOrCreate()
-for compatibility with older runtimes. With a new runtime, it's
-equivalent to DatabricksSession.builder.getOrCreate().
-*/ -}}
 from pyspark.sql import SparkSession
 
-def get_taxis():
-  spark = SparkSession.builder.getOrCreate()
+def get_taxis(spark: SparkSession):
   return spark.read.table("samples.nyctaxi.trips")
 
 def main():
-  get_taxis().show(5)
+  from databricks.connect import DatabricksSession as SparkSession
+  spark = SparkSession.builder.getOrCreate()
```

Comment:
This unfortunately won't work in any older runtime, so we can't use it in main.py. It wouldn't be a good customer experience if our template didn't work out of the box on older runtimes. So I think we can only apply this workaround in the testing code. 

An alternative might be to rely on the existence of the `spark` global but that's going to be another ugly workaround. 

Ideally the standard `SparkSession` would just work here 🤷

Reply:
This won't work with older runtimes. We can either chose not to support them or have a more verbose workaround

Reply:
As discussed this even breaks DLT!

Reply:
The DLT example would still work since we are calling `get_taxis()` directly from the DLT notebook but it's correct that this specific code is not supported by DLT.

Reply:
You wanted to do a try/except here right?

Reply:
yes, we'll go for the try/catch